Pr/21 lps 47199

arbetts opened the pull request.

arbetts committed:
LPS-47199 owner's are unable to view their journalArticle drafts from the home tab in the control panel

arbetts committed:
LPS-47199 also fix the "recent" tab so that the owner's draft articles show up

hhuijser committed:
LPS-47199 Source formatting

ealonso committed:
LPS-47199 The title url should be the same as action editURL

ealonso committed:
LPS-47199 Use a boolean instead of a new userId

juliocamarero committed:
LPS-47199 SF

ealonso committed:
LPS-47199 Adds new fields includeOwner and userId to QueryDefinition

ealonso committed:
LPS-47199 Support ownership in CustomSQL

ealonso committed:
LPS-47199 Uses the query definition for userId and includeOwner

arbetts committed:
LPS-47199 rename methods back to the parameters they use since userId is included in the queryDefinition

arbetts committed:
LPS-47199 define contract between CustomSQL.java and QueryDefinition.java with test

arbetts committed:
LPS-47199 fix JournalArticleFinderTest errors and add test case for owned draft articles

arbetts committed:
LPS-47199 wiki finder methods to search by owner and status

arbetts committed:
LPS-47199 apply to jsp

arbetts committed:
LPS-47199 test coverage for new finder method

brianchandotcom committed:
LPS-47199 SF

brianchandotcom committed:
LPS-47199 SF

brianchandotcom committed:
LPS-47199 SF

brianchandotcom committed:
LPS-47199 SF

brianchandotcom committed:
LPS-47199 Sort

brianchandotcom committed:
LPS-47199 Jim, please readd.

brianchandotcom committed:
LPS-47199 Sort

brianchandotcom committed:
LPS-47199 Sort

arbetts committed:
LPS-47199 rename test methods after JournalArticleFinderTest pattern and use random test util for string generation in test

arbetts committed:
LPS-47199 rename to disambiguate; clarify relationship between userId and includeOwner in QueryDefinition, and use USER_ID instead of OWNER in sql

arbetts committed:
LPS-47199 SF

arbetts committed:
LPS-47199 ant build-service in journal and wiki modules

arbetts committed:
LPS-47199 update test

liferay-continuous-integration commented:
<h1>Some tests FAILED!</h1><p>Build Time: 53 minutes 46 seconds</p><h4>Job Results:</h4><p>3 Jobs Passed.<br />2 Jobs Failed.</p><h1>See the following links for more information.</h1><ul><li><h3><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest(master)/537/">test-portal-acceptance-pullrequest(master)/537</a></h3><pre>Completed with the status of FAILURE.</pre></li><li><h3><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-backend(master)/398/">test-portal-acceptance-pullrequest-backend(master)/398</a></h3><h4>2 Failures.</h4><ul><li><strong><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-backend%28master%29/AXIS_VARIABLE=37,label_exp=!master/398/">test-portal-acceptance-pullrequest-backend(master) ?? 37,!master #398</a></strong><pre>e&#40;Echo.java:66&#41;<br />&#09;at org.apache.tools.ant.UnknownElement.execute&#40;UnknownElement.java:292&#41;<br />&#09;at sun.reflect.GeneratedMethodAccessor4.invoke&#40;Unknown Source&#41;<br />&#09;at sun.reflect.DelegatingMethodAccessorImpl.invoke&#40;DelegatingMethodAccessorImpl.java:43&#41;<br />&#09;at java.lang.reflect.Method.invoke&#40;Method.java:606&#41;<br />&#09;at org.apache.tools.ant.dispatch.DispatchUtils.execute&#40;DispatchUtils.java:106&#41;<br />&#09;at org.apache.tools.ant.Task.perform&#40;Task.java:348&#41;<br />&#09;at org.apache.tools.ant.taskdefs.Sequential.execute&#40;Sequential.java:68&#41;<br />&#09;at net.sf.antcontrib.logic.IfTask.execute&#40;IfTask.java:197&#41;<br />&#09;at sun.reflect.NativeMethodAccessorImpl.invoke0&#40;Native Method&#41;<br />&#09;at sun.reflect.NativeMethodAccessorImpl.invoke&#40;NativeMethodAccessorImpl.java:57&#41;<br />&#09;at sun.reflect.DelegatingMethodAccessorImpl.invoke&#40;DelegatingMethodAccessorImpl.java:43&#41;<br />&#09;at java.lang.reflect.Method.invoke&#40;Method.java:606&#41;<br />&#09;at org.apache.tools.ant.dispatch.DispatchUtils.execute&#40;DispatchUtils.java:106&#41;<br />&#09;at org.apache.tools.ant.TaskAdapter.execute&#40;TaskAdapter.java:154&#41;<br />&#09;at org.apache.tools.ant.UnknownElement.execute&#40;UnknownElement.java:292&#41;<br />&#09;at sun.reflect.GeneratedMethodAccessor4.invoke&#40;Unknown Source&#41;<br />&#09;at sun.reflect.DelegatingMethodAccessorImpl.invoke&#40;DelegatingMethodAccessorImpl.java:43&#41;<br />&#09;at java.lang.reflect.Method.invoke&#40;Method.java:606&#41;<br />&#09;at org.apache.tools.ant.dispatch.DispatchUtils.execute&#40;DispatchUtils.java:106&#41;<br />&#09;at org.apache.tools.ant.Task.perform&#40;Task.java:348&#41;<br />&#09;at org.apache.tools.ant.taskdefs.Sequential.execute&#40;Sequential.java:68&#41;<br />&#09;at net.sf.antcontrib.logic.IfTask.execute&#40;IfTask.java:197&#41;<br />&#09;at sun.reflect.NativeMethodAccessorImpl.invoke0&#40;Native Method&#41;<br />&#09;at sun.reflect.NativeMethodAccessorImpl.invoke&#40;NativeMethodAccessorImpl.java:57&#41;<br />&#09;at sun.reflect.DelegatingMethodAccessorImpl.invoke&#40;DelegatingMethodAccessorImpl.java:43&#41;<br />&#09;at java.lang.reflect.Method.invoke&#40;Method.java:606&#41;<br />&#09;at org.apache.tools.ant.dispatch.DispatchUtils.execute&#40;DispatchUtils.java:106&#41;<br />&#09;at org.apache.tools.ant.TaskAdapter.execute&#40;TaskAdapter.java:154&#41;<br />&#09;at org.apache.tools.ant.UnknownElement.execute&#40;UnknownElement.java:292&#41;<br />&#09;at sun.reflect.GeneratedMethodAccessor4.invoke&#40;Unknown Source&#41;<br />&#09;at sun.reflect.DelegatingMethodAccessorImpl.invoke&#40;DelegatingMethodAccessorImpl.java:43&#41;<br />&#09;at java.lang.reflect.Method.invoke&#40;Method.java:606&#41;<br />&#09;at org.apache.tools.ant.dispatch.DispatchUtils.execute&#40;DispatchUtils.java:106&#41;<br />&#09;at org.apache.tools.ant.Task.perform&#40;Task.java:348&#41;<br />&#09;at org.apache.tools.ant.taskdefs.Sequential.execute&#40;Sequential.java:68&#41;<br />&#09;at org.apache.tools.ant.UnknownElement.execute&#40;UnknownElement.java:292&#41;<br />&#09;at sun.reflect.GeneratedMethodAccessor4.invoke&#40;Unknown Source&#41;<br />&#09;at sun.reflect.DelegatingMethodAccessorImpl.invoke&#40;DelegatingMethodAccessorImpl.java:43&#41;<br />&#09;at java.lang.reflect.Method.invoke&#40;Method.java:606&#41;<br />&#09;at org.apache.tools.ant.dispatch.DispatchUtils.execute&#40;DispatchUtils.java:106&#41;<br />&#09;at org.apache.tools.ant.Task.perform&#40;Task.java:348&#41;<br />&#09;at org.apache.tools.ant.taskdefs.MacroInstance.execute&#40;MacroInstance.java:396&#41;<br />&#09;at org.apache.tools.ant.UnknownElement.execute&#40;UnknownElement.java:292&#41;<br />&#09;at sun.reflect.GeneratedMethodAccessor4.invoke&#40;Unknown Source&#41;<br />&#09;at sun.reflect.DelegatingMethodAccessorImpl.invoke&#40;DelegatingMethodAccessorImpl.java:43&#41;<br />&#09;at java.lang.reflect.Method.invoke&#40;Method.java:606&#41;<br />&#09;at org.apache.tools.ant.dispatch.DispatchUtils.execute&#40;DispatchUtils.java:106&#41;<br />&#09;at org.apache.tools.ant.Task.perform&#40;Task.java:348&#41;<br />&#09;at org.apache.tools.ant.taskdefs.Sequential.execute&#40;Sequential.java:68&#41;<br />&#09;at org.apache.tools.ant.UnknownElement.execute&#40;UnknownElement.java:292&#41;<br />&#09;at sun.reflect.GeneratedMethodAccessor4.invoke&#40;Unknown Source&#41;<br />&#09;at sun.reflect.DelegatingMethodAccessorImpl.invoke&#40;DelegatingMethodAccessorImpl.java:43&#41;<br />&#09;at java.lang.reflect.Method.invoke&#40;Method.java:606&#41;<br />&#09;at org.apache.tools.ant.dispatch.DispatchUtils.execute&#40;DispatchUtils.java:106&#41;<br />&#09;at org.apache.tools.ant.Task.perform&#40;Task.java:348&#41;<br />&#09;at org.apache.tools.ant.taskdefs.MacroInstance.execute&#40;MacroInstance.java:396&#41;<br />&#09;at org.apache.tools.ant.UnknownElement.execute&#40;UnknownElement.java:292&#41;<br />&#09;at sun.reflect.GeneratedMethodAccessor4.invoke&#40;Unknown Source&#41;<br />&#09;at sun.reflect.DelegatingMethodAccessorImpl.invoke&#40;DelegatingMethodAccessorImpl.java:43&#41;<br />&#09;at java.lang.reflect.Method.invoke&#40;Method.java:606&#41;<br />&#09;at org.apache.tools.ant.dispatch.DispatchUtils.execute&#40;DispatchUtils.java:106&#41;<br />&#09;at org.apache.tools.ant.Task.perform&#40;Task.java:348&#41;<br />&#09;at org.apache.tools.ant.Target.execute&#40;Target.java:435&#41;<br />&#09;at org.apache.tools.ant.Target.performTasks&#40;Target.java:456&#41;<br />&#09;at org.apache.tools.ant.Project.executeSortedTargets&#40;Project.java:1393&#41;<br />&#09;at org.apache.tools.ant.Project.executeTarget&#40;Project.java:1364&#41;<br />&#09;at org.apache.tools.ant.helper.DefaultExecutor.executeTargets&#40;DefaultExecutor.java:41&#41;<br />&#09;at org.apache.tools.ant.Project.executeTargets&#40;Project.java:1248&#41;<br />&#09;at org.apache.tools.ant.Main.runBuild&#40;Main.java:851&#41;<br />&#09;at org.apache.tools.ant.Main.startAnt&#40;Main.java:235&#41;<br />&#09;at org.apache.tools.ant.launch.Launcher.run&#40;Launcher.java:280&#41;<br />&#09;at org.apache.tools.ant.launch.Launcher.main&#40;Launcher.java:109&#41;</pre></li><li><strong><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-backend%28master%29/AXIS_VARIABLE=6,label_exp=!master/398/">test-portal-acceptance-pullrequest-backend(master) ?? 6,!master #398</a></strong><pre>start-current-job:<br /><br />BUILD SUCCESSFU</pre></li></ul></li><li><h3><a href="https://test-1-4.liferay.com/job/test-portal-acceptance-pullrequest-plugins(master)/432/">test-portal-acceptance-pullrequest-plugins(master)/432</a></h3></li><li><h3><a href="https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-portalweb(master)/264/">test-portal-acceptance-pullrequest-portalweb(master)/264</a></h3></li><li><h3><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-source(master)/504/">test-portal-acceptance-pullrequest-source(master)/504</a></h3></li></ul>

arbetts commented:
ci:retest

liferay-continuous-integration commented:
<h1>All tests PASSED!</h1><p>Build Time: 54 minutes 29 seconds</p><h4>Job Results:</h4><p>5 Jobs Passed.<br />0 Jobs Failed.</p><h1>See the following links for more information.</h1><ul><li><h3><a href="https://test-1-1.liferay.com/job/test-portal-acceptance-pullrequest(master)/554/">test-portal-acceptance-pullrequest(master)/554</a></h3></li><li><h3><a href="https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-backend(master)/404/">test-portal-acceptance-pullrequest-backend(master)/404</a></h3></li><li><h3><a href="https://test-1-6.liferay.com/job/test-portal-acceptance-pullrequest-plugins(master)/409/">test-portal-acceptance-pullrequest-plugins(master)/409</a></h3></li><li><h3><a href="https://test-1-5.liferay.com/job/test-portal-acceptance-pullrequest-portalweb(master)/280/">test-portal-acceptance-pullrequest-portalweb(master)/280</a></h3></li><li><h3><a href="https://test-1-2.liferay.com/job/test-portal-acceptance-pullrequest-source(master)/524/">test-portal-acceptance-pullrequest-source(master)/524</a></h3></li></ul>

brianchandotcom commented:
@arbetts Can you resend:

1.) Regarding CustomSQLTest, we've never had a pattern where we set fields in a constructor like this. We do it via a static setUpClass. Also, if you set things up like that, you need to undo them. These tests run across the cluster and are shared.

2.) Since QueryDefinition expects "ownerUserId"

```
if (queryDefinition.getOwnerUserId() > 0) {
			if (queryDefinition.isIncludeOwner()) {
				StringBundler sb = new StringBundler();

				sb.append(StringPool.OPEN_PARENTHESIS);
				sb.append(tableName);
				sb.append(_USER_ID_CONDITION_DEFAULT);
				sb.append(" AND ");
				sb.append(tableName);
				sb.append(_STATUS_CONDITION_INVERSE);
				sb.append(StringPool.CLOSE_PARENTHESIS);

				sql = sql.replace(_USER_ID_KEYWORD, sb.toString());

				sql = sql.replace(_USER_ID_AND_OR_CONNECTOR, " OR ");
```

The logic in the sql xml should look for [$OWNER_USER_ID$]. Vars should go with it as well. We have to do this, or else we may run into a case where there is already a [$USER_ID$] in the sql statement.

3.) long ownerUserId, boolean includeOwner. Should it be includeOwnerUser or includeOwnerUserId?

I just need an argument one way or another. I'm leaning toward includeOwner too, but I want to hear an argument from you for why...

Let's go over this in person. Come grab me when you send this next revision.

Thx!

arbetts commented:
@brianchandotcom 

1) updated test to use static setup and tear down 

2) updated xml to use OWNER_USER_ID to lessen the likelihood of conflicts

3) I think it should be includeOwner because it conveys the idea without being too cumbersome, though none of the names stand out as winners. To me, an owner is also a user so the name inlcudeOwnerUser seems redundant.

https://github.com/brianchandotcom/liferay-portal/pull/28451
